src/gui/SerialConfig.cpp

```diff
@@ -0,0 +1,59 @@
+#include "gui/SerialConfig.hpp"
+
+#include <imgui.h>
+
+
+using namespace gui;
+
+
+SerialConfig::SerialConfig(hwio::SerialProto& serialProto) :
+    _ports(nullptr),
+    _port_idx(0),
+    _port_baud(576000),
+    _serialProto(serialProto)
+{
+    sp_list_ports(&_ports);
+}
+
+SerialConfig::~SerialConfig()
+{
+}
+
+void SerialConfig::render()
+{
+    int err;
+
+    ImGui::Begin("Serial Config");
```

Comment:
The windows initially open stacked, potentially undersized and they might get moved off the initial window dimensions so maybe add a crude fix until we figure out the rest of the gui?
```
    ImGui::SetNextWindowPos(ImVec2(10,10), ImGuiSetCond_Once);
    ImGui::SetNextWindowSize(ImVec2(250,125), ImGuiSetCond_Once);
    ImGui::SetNextWindowCollapsed(true, ImGuiSetCond_Once);
    ImGui::Begin("Serial Config");
```